### PR TITLE
feat(restify): instrumentation

### LINKED ISSUE #440: Add support for the restify framework
[restify](http://restify.com/) is a popular framework for building REST API's in Node.js. To improve the out-of-the-box experience of the Elastic APM Node.js agent, it would make sense to hook into their router to automatically name request transactions based on the restify route name. Most other APM vendors ship with support for restify already.

Here is it's usage compared to other popular frameworks:

| Framework | Weekly Downloads | Supported |
|------------|--------------------|-----------|
| Express | 3,679,726 | ✅ |
| hapi | 145,670 | ✅ |
| Koa | 128,837 | ✅ |
| restify | 58,605 |
| Sails | 19,280 |

### PR DESCRIPTION
**Pull request by Qard**
Fixes elastic/apm-agent-nodejs#440
Depends on elastic/apm-agent-nodejs#521

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation

### CODE REVIEW

**Comment by Qard** [comment]
Error reporting currently only works with 7.x. I'll see if I can get older versions working tomorrow.

**Comment by elasticdog** [comment]
The Jenkins worker froze up for some reason during these tests, so I aborted the run. Kicking it off again for you...

Jenkins, test this please.

**Comment by Qard** [comment]
Jenkins is not being very cooperative right now...

Jenkins, test this please.

**Comment by watson** [comment]
Jenkins, test this please

File: test/instrumentation/modules/restify.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  t.on('end', done)
+
+  server.get('/hello/:name', (req, res, next) => {
+    res.send({
+      message: 'hello ' + req.params.name
+    })
+    next()
+  })
+
+  server.listen(null, '0.0.0.0', function () {
```
[the remark this answers is not in the document]
Comment (watson) [reply test/instrumentation/modules/restify.js]: I'm 👍 with merging this after adding these comments btw

[the remark this answers is not in the document]
Comment (Qard) [reply test/instrumentation/modules/restify.js]: fixed

**Pull request merged**